File: app/domain/entities/tipo_impuesto_distribucion.py
from dataclasses import dataclass, field
from typing import Optional
from app.domain.entities.enums import TipoDistribucionImpuestoEnum, TipoReparticionBaseImpuestoEnum

@dataclass
class TipoImpuestoEtiqueta:
    id: Optional[int] = None
    nombre: str = ""
    descripcion: Optional[str] = None

@dataclass
class TipoImpuestoDistribucion:
    id: Optional[int] = None
    tipo_impuesto_id: Optional[int] = None
    tipo_reparticion: TipoDistribucionImpuestoEnum = TipoDistribucionImpuestoEnum.FACTURA
    factor_porcentaje: float = 100.0
    basado_en: TipoReparticionBaseImpuestoEnum = TipoReparticionBaseImpuestoEnum.IMPUESTO
    etiqueta_id: Optional[int] = None
    cuenta_contable: Optional[str] = None
    
    # Relaciones opcionales (cargadas por el repo)
    etiqueta: Optional[TipoImpuestoEtiqueta] = None

File: frontend/src/domain/entities/TipoImpuestoDistribucion.ts
export enum TipoDistribucionImpuestoEnum {
    FACTURA = 'factura',
    REEMBOLSO = 'reembolso'
}

export enum TipoReparticionBaseImpuestoEnum {
    BASE = 'base',
    IMPUESTO = 'impuesto'
}

export interface TipoImpuestoEtiqueta {
    id: number;
    nombre: string;
    descripcion?: string;
}

export interface TipoImpuestoDistribucion {
    id?: number;
    tipo_impuesto_id?: number;
    tipo_reparticion: TipoDistribucionImpuestoEnum;
    factor_porcentaje: number;
    basado_en: TipoReparticionBaseImpuestoEnum;
    etiqueta_id?: number;
    cuenta_contable?: string;
    etiqueta?: TipoImpuestoEtiqueta;
}
